### PR TITLE
AN-158 Add handling for notices in Gutenberg

### PR DESCRIPTION
**Pull request by kevinfodness**
* Add handler for notices in Gutenberg.
* Disable normal notice handling when Gutenberg is active.
* General cleanup/organization from the last round of changes.

### CODE REVIEW

File: admin/class-admin-apple-news.php [hunk 1]
```diff
@@ -240,9 +256,13 @@ public static function get_image_sizes() {
 	 * @access public
 	 */
 	public static function show_error( $message ) {
-		echo '<div class="apple-news-notice apple-news-notice-error"><p>'
-			. esc_html( $message )
-			. '</p></div>';
+		if ( apple_news_block_editor_is_active_for_post() ) {
+			Admin_Apple_Notice::error( $message );
+		} else {
+			echo '<div class="apple-news-notice apple-news-notice-error"><p>'
```
Comment (pattiereaves) [inline admin/class-admin-apple-news.php]: Would role=alert make sense here?

File: assets/js/pluginsidebar/components/notifications/index.js [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+    const {
+      notifications,
+    } = this.state;
+
+    const visibleNotifications = notifications
+      .filter((notification) => true !== notification.dismissed);
+
+    return (
+      <Fragment>
+        {visibleNotifications.map((notification) => (
```
Comment (pattiereaves) [inline assets/js/pluginsidebar/components/notifications/index.js]: Would this make sense as an unordered list?

File: assets/js/pluginsidebar/components/sidebar/index.js [hunk 3]
```diff
@@ -216,20 +220,22 @@ class Sidebar extends React.PureComponent {
...
-                key={id}
-                label={name}
-                checked={- 1 !== selectedSectionsArray.indexOf(id)}
-                onChange={
-                  (checked) => this.updateSelectedSections(checked, id)
-                }
-              />
-            ))
-          }
+          {Array.isArray(sections) && sections.map(({ id, name }) => (
```
Comment (pattiereaves) [inline assets/js/pluginsidebar/components/sidebar/index.js]: Would this make sense as an unordered list?

File: admin/class-admin-apple-news.php (hunk 1 above)
Comment (kevinfodness) [reply admin/class-admin-apple-news.php]: Yep!

File: assets/js/pluginsidebar/components/notifications/index.js (hunk 2 above)
Comment (kevinfodness) [reply assets/js/pluginsidebar/components/notifications/index.js]: I thought about that. However, it's unlikely that there is going to be more than one alert in the vast majority of circumstances, and I'd have to undo a bunch of built-in list formatting to get the built-in styles for the Notice component to display properly.

File: assets/js/pluginsidebar/components/sidebar/index.js (hunk 3 above)
Comment (kevinfodness) [reply assets/js/pluginsidebar/components/sidebar/index.js]: Here, it probably does, yes.

**Pull request merged**